### DISCUSSION TITLE
Model in which flow is prescribed by geostrophic balance

### DISCUSSION THREAD

**Discussion by mkrapp** [CliMA/Oceananigans.jl#2823] [Computational science!] [answered]
Hi, 

First of all, thank you for [Oceananigans.jl](https://github.com/CliMA/Oceananigans.jl). It a fantastic piece of work.

I would like to run a `NonhydrostaticModel` simulation with a flow with amplitude $U_0$, which should be achieved through forcing with a constant pressure gradient in the y-direction. I want the parcel of ocean to be in geostrophic balance, i.e., $\frac{1}{\rho_O}p_y = −fU_0$, with `f=-1.37e-4` in my example.

First, I thought this can be simply achieved using a `BackgroundField` and passing it as `background_fields = (; u = u_bg)` but the results don't make sense. Everything seems to move in the y-direction with no effect of Coriolis. 

I couldn't find any examples using the pressure fields as forcing/background. Any ideas suggestions?

**Comment by glwagner** [comment, marked as answer]
Can you be more specific about what you are trying to simulate?

With `background_fields = (; u = u_bg)`, where `u_bg` is just a positive constant, then you'll simulate a situation in which everything moves to the right at a constant velocity. But this is probably not what you want, because it just amounts to a shift in the frame of reference.

If you impose an initial condition `set!(model, u=u0)` where `u0` is a constant, and you're using `coriolis=FPlane(1.0)`, then you'll obtain an inertial oscillation. Is that what you are looking for?

**Reply by mkrapp** [reply]
Hey Gregory. That's what I was looking for. 💯 Too obvious, obviously. Thanks heaps!!!
